edb/schema/types.py

```diff
@@ -2376,3 +2376,36 @@ def materialize_type_in_attribute(
         )
 
     return schema
+
+
+def is_type_compatible(
+    type_a: Type,
+    type_b: Type,
+    *,
+    schema: s_schema.Schema,
+) -> bool:
+
```

Comment:
Let's add a docstring here, explaining that this is needed to make EdgeQL implicit casts explicit in SQL, which has differing semantics.